### PR TITLE
MINOR: configurable package in codegen

### PR DESCRIPTION
**Pull request by vvcephei**
Adds the ability to specify what package the codegen will produce.

Extracted from #7248

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by vvcephei** [comment]
Hey @cmccabe , do you have time to give this one a look also?

It's a smaller change. Basically, in #7248 we're going to use the code generator for Streams also, and we're obviously not going to want to put the generated code in the `common` package.

I figured it makes sense to pass it in as an argument in the same place you specify the output directory, since the package name and output directory are related anyway. I decided not to do any fancypants parsing of the directory to automatically determine the package. I could be convinced to do it; it's just simpler this way, and it doesn't seem to hard to get the package name right.

Thanks,
-John

**Comment by vvcephei** [comment]
Failures unrelated

Retest this, please.

**Comment by vvcephei** [comment]
Oh boy,

Retest this, please.

File: generator/src/main/java/org/apache/kafka/message/HeaderGenerator.java [hunk 1]
```diff
@@ -45,26 +46,28 @@
         ""
     };
 
-    static final String PACKAGE = "org.apache.kafka.common.message";
 
     private final CodeBuffer buffer;
 
     private final TreeSet<String> imports;
+    private String packageName;
```
Comment (cmccabe) [inline generator/src/main/java/org/apache/kafka/message/HeaderGenerator.java]: should be `final`

File: generator/src/test/java/org/apache/kafka/message/MessageDataGeneratorTest.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ public void testNullDefaults() throws Exception {
                 "      \"nullableVersions\": \"2+\", \"default\": \"null\" }",
                 "  ]",
                 "}")), MessageSpec.class);
-        new MessageDataGenerator().generate(testMessageSpec);
+        new MessageDataGenerator("test").generate(testMessageSpec);
```
Comment (cmccabe) [inline generator/src/test/java/org/apache/kafka/message/MessageDataGeneratorTest.java]: This should be `org.apache.kafka.common.message` to match the directory that the generated source files are outputted to

File: generator/src/test/java/org/apache/kafka/message/MessageDataGeneratorTest.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ public void testInvalidNullDefaultForInt() throws Exception {
             "  ]",
             "}")), MessageSpec.class);
         try {
-            new MessageDataGenerator().generate(testMessageSpec);
+            new MessageDataGenerator("test").generate(testMessageSpec);
```
Comment (cmccabe) [inline generator/src/test/java/org/apache/kafka/message/MessageDataGeneratorTest.java]: This should be `org.apache.kafka.common.message` to match the directory that the generated source files are outputted to

File: generator/src/test/java/org/apache/kafka/message/MessageDataGeneratorTest.java [hunk 4]
```diff
@@ -83,7 +83,7 @@ public void testInvalidNullDefaultForPotentiallyNonNullableArray() throws Except
                 "  ]",
                 "}")), MessageSpec.class);
         try {
-            new MessageDataGenerator().generate(testMessageSpec);
+            new MessageDataGenerator("test").generate(testMessageSpec);
```
Comment (cmccabe) [inline generator/src/test/java/org/apache/kafka/message/MessageDataGeneratorTest.java]: This should be `org.apache.kafka.common.message` to match the directory that the generated source files are outputted to

**Comment by cmccabe** [comment]
looks good overall.  I left a few comments.

**Comment by vvcephei** [comment]
Thanks for the review, @cmccabe !

I've updated the PR and addressed your comments.

**Comment by vvcephei** [comment]
Retest this, please.

**Comment by vvcephei** [comment]
Thanks, @cmccabe !

**Pull request merged**